### ISSUE TITLE
no-trailing-whitespace should, optionally, NOT ignore multiline template strings

### ISSUE THREAD

**Issue by mpsijm** [eslint/eslint#14183]
**What rule do you want to change?** `no-trailing-space`

**Does this change cause the rule to produce more or fewer warnings?** More, optionally

**How will the change be implemented? (New option, new default behavior, etc.)?** New option, name could be `ignoreTemplateStrings`, with default `true`, because most people (see #5786) use template strings in a way where the extra spacing is significant.

**Please provide some example code that this change will affect:**

```js
import cntl from "cntl";

const buttonClassName = cntl`
    rounded    
    border
    py-2
    w-full
    mx-auto
    justify-center
`;
```
(This code uses [cntl](https://www.npmjs.com/package/cntl) and [Tailwind](https://tailwindcss.com/)

**What does the rule currently do for this code?** Not give a warning/error on the trailing spaces after `rounded`

**What will the rule do after it's changed?** Give a warning/error on the trailing spaces after `rounded`

**Are you willing to submit a pull request to implement this change?** I could try :slightly_smiling_face:

**Comment by mdjermanovic** [comment]
Hi @mpsijm, thanks for the issue!

The team has made the difficult decision to [freeze stylistic rules](https://eslint.org/blog/2020/05/changes-to-rules-policies), so we won't be adding new options to the `no-trailing-spaces` rule.

You can configure the [no-restricted-syntax](https://eslint.org/docs/rules/no-restricted-syntax) rule to report trailing spaces in template literals, with the following selector: 

```
TemplateElement[value.raw=/[ \\t]\\n/]
```

[Online Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtc3ludGF4OiBbXCJlcnJvclwiLCB7XG4gICAgc2VsZWN0b3I6IFwiVGVtcGxhdGVFbGVtZW50W3ZhbHVlLnJhdz0vWyBcXFxcdF1cXFxcbi9dXCIsXG4gICAgbWVzc2FnZTogXCJUcmFpbGluZyBzcGFjZShzKSBmb3VuZCBpbiBhIHRlbXBsYXRlIGxpdGVyYWwuXCJcbn1dICovXG5cbmBzcGFjZSBoZXJlIC0+IFxuYDtcblxuYG5vIHRyYWlsaW5nIHNwYWNlc1xuYDsiLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjEyLCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

I'm closing this issue per our [rules policies](https://github.com/eslint/eslint#stylistic-rule-updates).
